stairway/src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -95,13 +95,26 @@ boolean isEmpty() {
   }
 
   /**
-   * Return the object from the hash map deserialized to the right type. Throw an exception if the
-   * Object cannot be deserialized to that type.
+   * Returns true if this map contains a mapping of the passed key.
+   *
+   * @param key to look up in the map
+   * @return true if present
+   */
+  boolean containsKey(String key) {
```

Comment:
Should we make this public? I think we can't distinguish null value from not-found in the FlightMap right now.

Reply:
I was thinking that maybe a missing key should be an error instead of returning `null`. It seems reasonable to expect users to ensure that a key is always set before it's retrieved. If they want to handle the unset case, they can always catch the `KeyNotFound` exception.

Reply:
My original aim was to keep the interface compatible with Map. We are a bit afield of that now, so we could consider this. It would not be upward compatible.

Reply:
You should run it by more people than just me, but I think it's better to throw than return `null`. The map is used to pass data between steps, and I think it should be an error for a step to depend on a key that isn't always present.

Reply:
> Should we make this public? I think we can't distinguish null value from not-found in the FlightMap right now.

Good catch, this was meant to be public.

> You should run it by more people than just me, but I think it's better to throw than return null.

This sounds reasonable to me, but I'd rather not lump it into this effort, as it's already turned up a few more unexpected behaviors than we would have liked...